src/modules/Elsa.Workflows.Core/Services/VariablePersistenceManager.cs

```diff
@@ -34,6 +34,9 @@ public async Task LoadVariablesAsync(WorkflowExecutionContext workflowExecutionC
                 if (driver == null)
                     continue;
 
+                if (excludeValueRetrievalForLargeDataDrivers && driver.Tags.Contains(StorageDriverTag.LargeData))
+                    continue;
```

Comment:
When you replace the `excludeValueRetrievalForLargeDataDrivers` parameter with a list of tags, you can replace this logic by simply checking to see if the driver contains all of the specified tags.